java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -1097,15 +1106,44 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_makeCudfColumnView(
       cudf::column_view data_column(cudf::data_type{cudf::type_id::INT8}, j_data_size, data);
       ret.reset(new cudf::column_view(cudf::data_type{cudf::type_id::STRING}, size, nullptr, valid,
                                       j_null_count, 0, {offsets_column, data_column}));
-    } else {
-      ret.reset(new cudf::column_view(n_data_type, size, data, valid, j_null_count));
+    } else if (n_type == cudf::type_id::LIST) {
+         JNI_NULL_CHECK(env, j_offset, "offset is null", 0);
+         cudf::column_view *child_view = reinterpret_cast<cudf::column_view *>(j_child);
+         cudf::size_type *offsets = reinterpret_cast<cudf::size_type *>(j_offset);
+         cudf::column_view offsets_column(cudf::data_type{cudf::type_id::INT32}, size + 1, offsets);
+         ret.reset(new cudf::column_view(cudf::data_type{cudf::type_id::LIST}, size, nullptr, valid,
+                                                 j_null_count, 0, {offsets_column, *child_view}));
+   } else {
+     ret.reset(new cudf::column_view(n_data_type, size, data, valid, j_null_count));
     }
 
     return reinterpret_cast<jlong>(ret.release());
   }
   CATCH_STD(env, 0);
 }
 
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_getChildColumnView(JNIEnv *env,
```

Comment:
Is the intention here to get to the `column_view` of the list's child?

Reply:
yes

Reply:
I dont think it will be very useful in this patch but I kept it to see if this is worth keeping.

I use `getChildCvPointer` to do the child view pointer work on java side.

Reply:
I wonder if the following would be equivalent:
```c++
JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_getChildColumnView(
  JNIEnv *env,
  jobject j_object, jlong handle) {
  
  using namespace cudf;

  jni::auto_set_device(env);

  auto col {reinterpret_cast<column_view*>(handle)};
  return col->type().id() == type_id::LIST?
      new column_view(lists_column_view(col).child()) : 0;
}
```

The latest list code attempts to keep the child vectors intact, even for empty lists.
Also, I suspect that returning a default-constructed `column_view` would not be in keeping with the protocol of returning `nullptr` for invalid columns.

Reply:
I realize that this might be moot: `getChildColumnView()` should really take a child column index.

Reply:
Good point! Let me see what I can do here.

Reply:
Done.